### PR TITLE
fix: Better description to wait for two vol test

### PR DESCRIPTION
**Pull request by mateusoliveira43**
Add better description for why it is needed to have a wait in two volumes test case.

### CODE REVIEW

File: tests/e2e/backup_restore_suite_test.go [hunk 1]
```diff
@@ -143,9 +142,12 @@ func runBackupAndRestore(brCase BackupRestoreCase, expectedErr error, updateLast
 	nsRequiresResticDCWorkaround, err := NamespaceRequiresResticDCWorkaround(dpaCR.Client, brCase.ApplicationNamespace)
 	Expect(err).ToNot(HaveOccurred())
 
-	// TODO this should be a function, not an arbitrary sleep
-	log.Printf("Sleeping for %v to allow application to be ready for case %s", brCase.AppReadyDelay, brCase.Name)
-	time.Sleep(brCase.AppReadyDelay)
+	if strings.Contains(brCase.Name, "twovol") {
```
Comment (mpryc) [inline tests/e2e/backup_restore_suite_test.go:145]: I like this improvement of removing it from function, but the proper solution would be to something less "around x seconds", how about putting `readinessProbe` in the `tests/e2e/sample-applications/mysql-persistent/mysql-persistent-twovol-csi.yaml` ? 

```
            volumeMounts:
            - name: applog
              mountPath: /tmp/log/todoapp/
            readinessProbe:
              exec:
                command:
                - sh
                - -c
                - '[ -f /tmp/log/todoapp/app.log ]'
              initialDelaySeconds: 1    # Delay before the first probe
              periodSeconds: 5          # Probe interval
              timeoutSeconds: 1         # Maximum time for the probe to complete, should be immediate as it's just file check
              successThreshold: 1       # One time is enough to check for file 
              failureThreshold: 30      # Number of consecutive failed probes to be considered failed   
```

Comment (mateusoliveira43) [reply tests/e2e/backup_restore_suite_test.go:145]: That would check, until it succeeds, if the file exists, right?

I think that would not be enough, the problem we face is not that the file does not exist, it is that volume does not have all logs yet

example

first log is saying application started (this one creates files, if not present, I assume)
second log is saying 2 things were added to database (this one, without hardcoded timeout, is not present in backup)

Comment (mpryc) [reply tests/e2e/backup_restore_suite_test.go:145]: So we are checking for the actual item to be in database as well? That could be also part of the check here, just slightly different logic. I am thinking here to solve the issue once and for all to not come back to this problem and actually remove the logic from the go code as that should not be mixed too much. Once the container is ready (whatever we do to make it ready via init-container or probe) then the test should continue. The test itself should not have too much knowledge about application itself.

Comment (mateusoliveira43) [reply tests/e2e/backup_restore_suite_test.go:145]: would not that create a cyclic dependency? 
1. we need to check if application is running (also check probes)
2. then we interact with application (creating TO DO items)
3. then we backup

are you not asking to check 2 inside 1, which is impossible?

**Review by weshayutin — DISMISSED** [review]

File: tests/e2e/backup_restore_suite_test.go (hunk 1 above)
Comment (kaovilai) [reply tests/e2e/backup_restore_suite_test.go:145]: 2 is not inside 1. 1. only indicates app is awake and ready to accept commands.

**Review by kaovilai — DISMISSED** [review]

File: tests/e2e/lib/backup.go [hunk 2]
```diff
@@ -25,9 +24,6 @@ func CreateBackupForNamespaces(ocClient client.Client, veleroNamespace, backupNa
 			SnapshotMoveData:         &snapshotMoveData,
 		},
 	}
-	if snapshotVolumes {
```
Comment (mateusoliveira43) [inline tests/e2e/lib/backup.go:28]: @mrnold I reverted some changes from your PR that I think were not necessary (this option for example, is enabled by default)

Checking if the kubevirt test pass is enough to assure I did not break anything there?

Comment (mrnold) [reply tests/e2e/lib/backup.go:28]: Yes, that's fine. I wasn't 100% sure whether or not this was needed.

**Comment by mateusoliveira43** [comment]
13 azure, and 14 azure and aws had subscription test error
/retest

**Comment by mateusoliveira43** [comment]
/unhold
rebased in top of master (with fixed CI)

**Review by mpryc — APPROVED** [review]

File: tests/e2e/lib/backup.go [hunk 3]
```diff
@@ -9,11 +9,10 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/utils/pointer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func CreateBackupForNamespaces(ocClient client.Client, veleroNamespace, backupName string, namespaces []string, defaultVolumesToFsBackup bool, snapshotMoveData, snapshotVolumes bool) (velero.Backup, error) {
+func CreateBackupForNamespaces(ocClient client.Client, veleroNamespace, backupName string, namespaces []string, defaultVolumesToFsBackup bool, snapshotMoveData bool) (velero.Backup, error) {
```
Comment (mpryc) [inline tests/e2e/lib/backup.go:15]: Maybe simplify:

```
func CreateBackupForNamespaces(ocClient client.Client, veleroNamespace, backupName string, namespaces []string, defaultVolumesToFsBackup, snapshotMoveData bool) (velero.Backup, error) {}
```

File: tests/e2e/backup_restore_suite_test.go [hunk 4]
```diff
@@ -189,12 +186,16 @@ func runBackup(brCase BackupRestoreCase, backupName string) bool {
 	nsRequiresResticDCWorkaround, err := lib.NamespaceRequiresResticDCWorkaround(dpaCR.Client, brCase.Namespace)
 	gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
-	// TODO this should be a function, not an arbitrary sleep
-	log.Printf("Sleeping for %v to allow application to be ready for case %s", brCase.ReadyDelay, brCase.Name)
-	time.Sleep(brCase.ReadyDelay)
+	if strings.Contains(brCase.Name, "twovol") {
+		volumeSyncDelay := 30 * time.Second
```
Comment (mpryc) [inline tests/e2e/backup_restore_suite_test.go:190]: The 30 seconds is a guess ?

File: tests/e2e/lib/backup.go (hunk 3 above)
Comment (mateusoliveira43) [reply tests/e2e/lib/backup.go:15]: I hope this will be catch by future linter tasks

File: tests/e2e/backup_restore_suite_test.go (hunk 4 above)
Comment (mateusoliveira43) [reply tests/e2e/backup_restore_suite_test.go:190]: just used what was being used already

File: tests/e2e/lib/backup.go (hunk 3 above)
Comment (mateusoliveira43) [reply tests/e2e/lib/backup.go:15]: will update in linter task, not in this PR

**Review by mpryc — APPROVED** [review]

**Comment by mateusoliveira43** [comment]
/override ci/prow/4.12-e2e-test-aws
restore succeeds but application seems to unavailable after flake

**Pull request merged**